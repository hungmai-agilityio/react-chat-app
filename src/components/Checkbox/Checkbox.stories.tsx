import { Meta, StoryObj } from '@storybook/react';

// Components
import Checkbox from '@/components/Checkbox';

const meta: Meta<typeof Checkbox> = {
  component: Checkbox,
  tags: ['autodocs'],
  argTypes: {
    id: {
      description: 'Set id for checkbox'
    },
    children: {
      description: 'Render dynamic react node children '
    },
    isChecked: {
      description: 'Set checked'
    },
    onChange: {
      description: 'Handle the event when clicking on the button'
    }
  }
};

export default meta;

type Story = StoryObj<typeof Checkbox>;

export const NoCheck: Story = {
  args: {
    id: '701796f0-0253-11ef-804b-d90ac4375227',
    isChecked: false,
    children: 'No Check',
    onChange: () => {}
  }
};

export const Checked: Story = {
  args: {
    id: '701796f0-0253-11ef-804b-d90ac4375227',
    isChecked: true,
    children: 'No Check',
    onChange: () => {}
  }
};
