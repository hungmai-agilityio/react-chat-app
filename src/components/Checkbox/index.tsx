import { clsx } from 'clsx';
import { ReactNode } from 'react';

interface CheckboxProps {
  id: string;
  children: ReactNode;
  isChecked: boolean;
  onChange: (id: string) => void;
}

const Checkbox = ({ id, children, isChecked, onChange }: CheckboxProps) => {
  // Handle checkbox
  const handleChecked = () => {
    onChange(id);
  };

  return (
    <div
      id={id}
      className={clsx('pl-3 flex items-center gap-4 w-full', {
        'bg-blue-50': isChecked
      })}
    >
      <input
        type="checkbox"
        checked={isChecked}
        className="w-5 h-5 accent-primary"
        onChange={handleChecked}
      />
      <label onClick={handleChecked} className="w-full">
        {children}
      </label>
    </div>
  );
};

export default Checkbox;
