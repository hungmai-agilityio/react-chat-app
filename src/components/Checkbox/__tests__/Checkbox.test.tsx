import { fireEvent, render, screen } from '@testing-library/react';

// Components
import Checkbox from '@/components/Checkbox';

describe('Checkbox', () => {
  const propsDefault = {
    id: '701796f0-0253-11ef-804b-d90ac4375227',
    children: 'John',
    isChecked: false,
    onChange: jest.fn()
  };

  beforeEach(() => {
    jest.fn().mockClear();
  });

  it('render Checkbox when unchecked', () => {
    render(<Checkbox {...propsDefault} />);

    const element = screen.getByText(/John/);

    expect(element).toBeInTheDocument();
  });

  it('should call onChange event when checkbox is clicked', () => {
    render(<Checkbox {...propsDefault} />);

    const checkbox = screen.getByRole('checkbox');
    fireEvent.click(checkbox);

    expect(propsDefault.onChange).toHaveBeenCalledTimes(1);
    expect(propsDefault.onChange).toHaveBeenCalledWith(
      '701796f0-0253-11ef-804b-d90ac4375227'
    );
  });

  it('should be render match to snapshot when unchecked', () => {
    const { container } = render(<Checkbox {...propsDefault} />);

    expect(container).toMatchSnapshot();
  });

  it('should be render match to snapshot when checked', () => {
    const { container } = render(
      <Checkbox {...propsDefault} isChecked={true} />
    );

    expect(container).toMatchSnapshot();
  });
});
