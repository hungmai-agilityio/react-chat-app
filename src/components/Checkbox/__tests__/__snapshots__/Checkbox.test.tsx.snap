// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should be render match to snapshot when checked 1`] = `
<div>
  <div
    class="pl-3 flex items-center gap-4 w-full bg-blue-50"
    id="701796f0-0253-11ef-804b-d90ac4375227"
  >
    <input
      checked=""
      class="w-5 h-5 accent-primary"
      type="checkbox"
    />
    <label
      class="w-full"
    >
      John
    </label>
  </div>
</div>
`;

exports[`Checkbox should be render match to snapshot when unchecked 1`] = `
<div>
  <div
    class="pl-3 flex items-center gap-4 w-full"
    id="701796f0-0253-11ef-804b-d90ac4375227"
  >
    <input
      class="w-5 h-5 accent-primary"
      type="checkbox"
    />
    <label
      class="w-full"
    >
      John
    </label>
  </div>
</div>
`;
